packages/dd-trace/test/plugins/util/inferred_proxy.spec.js

```diff
@@ -0,0 +1,259 @@
+'use strict'
+
+require('../../setup/tap')
+
+const agent = require('../agent')
+const getPort = require('get-port')
+const { expect } = require('chai')
+const axios = require('axios')
+
+describe('Inferred Proxy Spans', function () {
+  let http
+  let appListener
+  let controller
+  let port
+
+  // tap was throwing timeout errors when trying to use hooks like `before`, so instead we just use this function
+  // and call before the test starts
+  const loadTest = async function (options) {
+    process.env.DD_SERVICE = 'aws-server'
+
+    port = await getPort()
+    require('../../../../dd-trace')
+
+    await agent.load(['http'], null, options)
+
+    http = require('http')
+
+    const server = new http.Server(async (req, res) => {
+      controller && await controller(req, res)
+      if (req.url === '/error') {
+        res.statusCode = 500
+        res.end(JSON.stringify({ message: 'ERROR' }))
+      } else {
+        res.writeHead(200)
+        res.end(JSON.stringify({ message: 'OK' }))
+      }
+    })
+
+    appListener = server.listen(port, '127.0.0.1')
+  }
+
+  // test cleanup function
+  const cleanupTest = function () {
+    appListener && appListener.close()
+    try {
+      agent.close({ ritmReset: false })
+    } catch {
+      // pass
+    }
+  }
+
+  const inferredHeaders = {
+    'x-dd-proxy': 'aws-apigateway',
+    'x-dd-proxy-request-time-ms': '1729780025473',
+    'x-dd-proxy-path': '/test',
+    'x-dd-proxy-httpmethod': 'GET',
+    'x-dd-proxy-domain-name': 'example.com',
+    'x-dd-proxy-stage': 'dev'
+  }
+
+  describe('without configuration', () => {
+    it('should create a parent span and a child span for a 200', async () => {
+      await loadTest({})
+
+      await axios.get(`http://127.0.0.1:${port}/`, {
+        headers: inferredHeaders
+      })
+
+      await agent.use(traces => {
+        for (const trace of traces) {
+          try {
+            const spans = trace
+
+            expect(spans.length).to.be.equal(2)
+
+            expect(spans[0]).to.have.property('name', 'aws.apigateway')
+            expect(spans[0]).to.have.property('service', 'example.com')
+            expect(spans[0]).to.have.property('resource', 'GET /test')
```

Comment:
There should also be a test to make sure the "type" of the span is also "web". If the `type` is in the `meta` section, that won't be enough for it to be extracted out.

Reply:
should be fixed